### DISCUSSION TITLE
single-file-format vs diffusers-dir-style format checkpoints, which one do you prefer?

### DISCUSSION THREAD

**Discussion by yiyixuxu** [huggingface/diffusers#7073] [General]
continue the discussion we started here https://github.com/huggingface/diffusers/pull/6487#issuecomment-1958790109

I'm asking for your preference - so this can and should be highly subjective:) feel free to express your opinions and explain the reasons if possible:)

**Comment by asomoza** [comment]
I think this should be differentiated between developers and users, people that just use the checkpoints would always prefer single file checkpoints since they're a lot easier to manage (manually).

For developers shouldn't  be that hard since is all done with code, I really don't care if I manage 1 file or 20 of them per model, what I do care is for efficiency and bandwidth, if I can somehow just load the `unet` while preserving the other modules that's a plus for me, if I can save more than 1GB of download and space for each model, that's also a plus.

Sadly there's no benefit today, since we really don't know what parts of a model we can reutilize or not, so I have to work with them as if all the modules are different which makes the `diffusers` format lose a lot of its value.

I'm guessing as the models gets bigger, people (especially hosting sites) would start to care more about multiple file checkpoints. One recent use case of multiple file format is the [SDXL-Lighting](https://huggingface.co/ByteDance/SDXL-Lightning) where they also released the `unets` alone since is the only part that changes between them.

Also is important to separate the HF HUB and the diffusers format, I often see people grouping them together, you can use the diffusers format or even huggingface hosting without the hub, the symlinks and the cache folder are part of the HUB and not of the file format. 

Maybe what we really need is something like the `nexus mod manager` but for SD checkpoints.
